### PR TITLE
Introduce skipInheritingClasses parameter

### PR DESCRIPTION
**Pull request by tjuchniewicz**
Fixes gh-635.

### CODE REVIEW

**Comment by tjuchniewicz** [comment]
Probably solves the same problem as https://github.com/kongchen/swagger-maven-plugin/pull/531

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java [hunk 1]
```diff
@@ -134,28 +134,32 @@
...
+            prefixes.add("");
         } else {
-            for (String location : locations) {
-                Set<Class<?>> c = new Reflections(location).getTypesAnnotatedWith(clazz, true);
-                classes.addAll(c);
+            prefixes.addAll(getLocations());
+        }
+        
+        for (String location : prefixes) {
+            Set<Class<?>> c = new Reflections(location).getTypesAnnotatedWith(clazz, true);
```
Comment (Postremus) [inline src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java]: If i am not mistaken, scanning with an empty string will search, basicly, everywhere for types?
Is this also the current behaviour, when no location is defined within the ```<locations></locations>``` element?

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java [hunk 2]
```diff
@@ -134,28 +134,32 @@
...
+            if (!skipInheritingClasses) {
                 Set<Class<?>> inherited = new Reflections(location).getTypesAnnotatedWith(clazz);
                 classes.addAll(inherited);
             }
         }
-
         return classes;
     }
-
+
```
Comment (Postremus) [inline src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java]: useless whitespace change

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java (hunk 1 above)
Comment (tjuchniewicz) [reply src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java]: Yes, behavior is the same as before.

File: src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java (hunk 2 above)
Comment (tjuchniewicz) [reply src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java]: Fixed.

**Comment by tjuchniewicz** [comment]
PR has been updated. Thanks @who !

**Pull request merged**